src/aggregate/aggregate_request.c

```diff
@@ -753,10 +742,8 @@ int AREQ_ApplyContext(AREQ *req, RedisSearchCtx *sctx, QueryError *status) {
   }
 
   ConcurrentSearchCtx_Init(sctx->redisCtx, &req->conc);
-  req->rootiter = QAST_Iterate(ast, opts, sctx, &req->conc, status);
-  if (!req->rootiter) {
-    return REDISMODULE_ERR;
-  }
+  req->rootiter = QAST_Iterate(ast, opts, sctx, &req->conc);
+  assert(req->rootiter);
```

Comment:
why?

Reply:
Look at QAST_Iterate there is no way it returns NULL, once again dead code ..

Reply:
We might eventually change QueryNode_EvalNode to contain an error. This is an API-level construct. As far as users of QAST_Iterate are concerned, it might return an error.